src/components/Movie.js

```diff
@@ -0,0 +1,115 @@
+import '../css/Movie.css'
+import React, { Component, Fragment } from 'react'
+import Loading from './Loading'
+import Error from './Error'
+import Trailer from '../components/Trailer'
+import MovieData from '../components/MovieData'
+import endpoints from '../endpoints'
+import star from '../assets/star.png'
+import dollarSign1 from '../assets/dollarSign1.png'
+import dollarSign2 from '../assets/dollarSign2.png'
+
+class Movie extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      movie: {},
+      errorStatus: '',
+      errorMessage: '',
+      loading: true
+    }
+  }
+
+  componentDidMount = () => {
+    fetch(`${endpoints.movies}/${this.props.id}`)
+      .then(response => {
+        if(!response.ok) {
+          throw new Error ({
+            status: response.status,
+            message: response.statusText
+          })
+        } return response.json()
+      })
+      .then(data => {
+        this.setState({
+          movie: data.movie,
+          loading: false
+        })
+      })
+      .catch(err => {
+        this.setState({
+          errorStatus: err.status,
+          errorMessage: err.message,
+          loading: false
+        })
+      })
+  }
+
+  render() {
+    const {
+      title,
+      poster_path,
+      backdrop_path,
+      release_date,
+      overview,
+      genres,
+      budget,
+      revenue,
+      runtime,
+      tagline,
+      average_rating
+    } = this.state.movie
+
+    return (
+      <>
+        {this.state.loading ? <Loading /> :
+          <>
+            {this.state.error ? <Error errorStatus={this.state.errorStatus} errorMessage={this.state.errorMessage} /> :
+              <>
+                <section className="movie-details">
+                  <img src={backdrop_path} alt="movie background" className="backdrop"/>
+                  <article className="text">
+                    <header className="header">
+                      <h1 className="title">{title}</h1>
+                        <div className="all-numbers">
+                          <div className="numbers">
+                            <div className="number-text">
+                              <h2 className="release">{release_date?.split('-')[0]}</h2>
```

Comment:
`?.` is called [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).  Optional chaining.